### ISSUE TITLE
Improve face detection

### ISSUE THREAD

**Issue by maxbbraun** [maxbbraun/fever#2]
Face detection currently only works in a limited way, because the detector was not made for thermal images. Two possible directions to take from here:

1. Use an RGB camera right next to the thermal camera, run face detection there, and reproject the detections into the thermal camera frame to read the temperature. Could use the [Raspberry Pi Camera Module V2](https://www.raspberrypi.org/products/camera-module-v2/) to keep it cheap and simple or a [Intel RealSense D400](https://www.intelrealsense.com/stereo-depth/) if #1 requires it.
   - _Pro:_ fairly straightforward implementation
   - _Con:_ additional hardware, reprojection introduces error

2. Make a thermal face detector. This could probably be done by fine-tuning a detector network previously trained on RGB data. [This dataset of 1557 thermal images with bounding boxes](https://github.com/maxbbraun/tdface-annotations) might be a start. As for the detector, maybe try [EfficientDet](https://github.com/google/automl/tree/master/efficientdet). In general, [TensorFlow Lite](https://www.tensorflow.org/lite) compatibility might be preferable so an [Edge TPU](https://coral.ai/products/accelerator/) can be used.
   - _Pro:_ clean all-in-one solution
   - _Con:_ more unknowns in implementation

In either case, #2 may also require face landmarks (eyes, nose, mouth) instead of plain bounding boxes for better accuracy.

See also:
- [Projection in RealSense SDK 2.0](https://github.com/IntelRealSense/librealsense/wiki/Projection-in-RealSense-SDK-2.0)

**Comment by maxbbraun** [comment]
I went with option 2 for now and created [Thermal Face](https://github.com/maxbbraun/thermal-face). It runs pretty well, but beware of [this issue](https://github.com/maxbbraun/thermal-face/issues/2).

**Comment by maxbbraun** [comment]
Issue #2 is closed.
